conan/tools/files/files.py

```diff
@@ -367,11 +373,22 @@ def print_progress(_, __):
                     output.error(f"Error extract {file_.filename}\n{str(e)}", error_type="exception")
         output.writeln("")
 
-
-def untargz(filename, destination=".", pattern=None, strip_root=False, extract_filter=None):
+def untargz(filename, destination=".", pattern=None, strip_root=False, extract_filter=None, output=None):
     # NOT EXPOSED at `conan.tools.files` but used in tests
     import tarfile
-    with tarfile.TarFile.open(filename, 'r:*') as tarredgzippedFile:
+    import io
+    class ProgressFileObject(io.FileIO):
```

Comment:
We already have a ``class FileProgress`` in the ``FileUploader``, maybe it is the time to extract it and reuse it. Or at least copy it as-is, and leave it for a later refactor. But it would be that at least the initial implementation is the same.

Reply:
I've extracted that `FileProgress` class and adapted it to different scenarios! See https://github.com/conan-io/conan/pull/17519/files#diff-9afbf22c7cefd3f7b642e9b1b988ba829d3c36e3092bfb6140d01ad0bba4c571R537